### ISSUE TITLE
TheHive Nodes

### ISSUE THREAD

**Issue by arkhelieldan** [n8n-io/n8n#1225]
Hi, first great Job for @naudouani for create these nodes and by team to implement.
![image](https://user-images.githubusercontent.com/71335495/101058671-221d0780-356c-11eb-9fe4-497df5b1431e.png)

Good, regarding thehive nodes, we have just only 3 types of artifacts, but by default thehive have all this artifacts:

  
```
autonomous-system
    domain
    file
    filename
    fqdn
    hash
    hostname
    ip
    mail
    mail-subject
    mail_subject
    other
    regexp
    registry
    uri_path
    url
    user-agent
```

Is it possible add manualy in nodes artifact type "**Hash**"for example, or need update this node in the next release to get these artifacts.

**Comment by janober** [comment]
Have honestly no experience with TheHive so can not say much about that. If that is correct and the other values should be added then we or anybody else can add them easily so that it gets released with the next version.

As a temporary workaround can you change the parameter to an expression and set it to whatever value you want.

**Comment by mluhta** [comment]
TheHive 3 has the following observable datatypes by default:

```
autonomous-system
domain
file
filename
fqdn
hash
ip
mail
mail_subject
other
regexp
registry
uri_path
url
user-agent
````

TheHive 4 has the following observable datatypes by default:

```
autonomous-system	
domain	
file	
filename	
fqdn	
hash	
hostname	
ip	
mail	
mail-subject	
other	
regexp	
registry	
uri_path	
url	
user-agent
```

So the difference between the versions is the ` hostname ` observable, which has been added in TheHive 4.
Users can also add their own observable datatypes, so the best way for this would be to load them from the TheHive instance.

TheHive 3 has API endpoint ` /api/list/list_artifactDataType ` and TheHive 4 has ` /api/observable/type?range=all `

It's apparently possible to load options "dynamically", so I guess we could do it that way by using the ` typeOptions.loadOptionsMethod `, right?

**Comment by janober** [comment]
Fix got released with n8n@0.115.0
